src/shell/interpreter.zig

```diff
@@ -1530,6 +1530,21 @@ pub fn NewInterpreter(comptime EventLoopKind: JSC.EventLoopKind) type {
                             }
 
                             if (this.word_idx >= this.node.atomsLen()) {
+                                if (this.node.hasTildeExpansion() and this.node.atomsLen() > 1) {
+                                    const homedir = this.base.shell.getHomedir();
+                                    defer homedir.deref();
+                                    if (this.current_out.items.len > 0) {
+                                        switch (this.current_out.items[0]) {
+                                            '/', '\\' => {
+                                                this.current_out.insertSlice(0, homedir.slice()) catch bun.outOfMemory();
+                                            },
+                                            else => {
+                                                // TODO: Handle username
+                                                this.current_out.insert(0, '~') catch bun.outOfMemory();
```

Comment:
for username maybe `bun.getUserName` can help

Reply:
Will do this later as `~username` not commonly used, and also implementation is complicated on windows.